PythonVisualizations/PriorityQueue.py

```diff
@@ -0,0 +1,340 @@
+        inside.  The position of the cell can either be an integer index in
+        the Array or the bounding box coordinates of the square.  If color
+        is not supplied, the next color in the palette is used.
+        An event handler is set up to update the VisualizationApp argument
+        with the cell's value if clicked with any button.
+        Returns the tuple, (square, text), of canvas items
+        """
+        # Determine position and color of cell
+        if isinstance(indexOrCoords, int):
+            rectPos = self.cellCoords(indexOrCoords)
+            valPos = self.cellCenter(indexOrCoords)
+        else:
+            rectPos = indexOrCoords
+            valPos = divide_vector(add_vector(rectPos[:2], rectPos[2:]), 2)
+        if color is None:
+            # Take the next color from the palette
+            color = drawable.palette[self.nextColor]
+            self.nextColor = (self.nextColor + 1) % len(drawable.palette)
+
+        cell_rect = self.canvas.create_rectangle(
+            *rectPos, fill=color, outline='', width=0)
+        cell_val = self.canvas.create_text(
+            *valPos, text=str(key), font=self.VALUE_FONT, fill=self.VALUE_COLOR)
+        handler = lambda e: self.setArgument(str(key))
+        for item in (cell_rect, cell_val):
+            self.canvas.tag_bind(item, '<Button>', handler)
+
+        return cell_rect, cell_val
+
+    def display(self):
+        self.canvas.delete("all")
+
+        self.index = self.createIndex(self.nItems, 'nItems', level=-1) # indicate priority        
+
+        for i in range(self.size):  # Draw grid of cells
+            self.createArrayCell(i)
+
+        # go through each Drawable in the list
+        for i in range(self.nItems):
+            drawItem = self.list[i]
+
+        self.window.update()
+        
+    def newArraySize(self, val):
+        callEnviron = self.createCallEnvironment()                
+        # Clear Array and reset size and list
+        self.size = val
+        self.list = [None]*self.size
+        self.nItems = 0
+        
+        self.display()
+        self.cleanUp(callEnviron)
+
+    # delete the last element of the queue, or None if empty
+    def remove(self):
+        callEnviron = self.createCallEnvironment()
+        if self.nItems == 0:
+            self.setMessage('Array is empty!')
+            return
+        self.startAnimations()
+        self.wait(0.3)
+
+        n = self.list[self.nItems - 1]
+
+        # Slide value rectangle up and off screen
+        items = (n.display_shape, n.display_val)
+        self.moveItemsOffCanvas(items, N, sleepTime=0.02)
+
+        self.nItems -= 1
+        self.moveItemsBy(self.index, (-self.CELL_SIZE, 0))  # move priority arrow
+
+        n = self.list.pop()
+
+        self.cleanUp(callEnviron)
+
+    def fixCells(self):       # Move canvas display items to exact cell coords
+        for i in range(self.nItems):
+            drawItem = self.list[i]
+        self.window.update()
```

Comment:
You fixed the loop in `fixCells()` to iterate over the first `nItems` in `self.list`, but it doesn't do anything to those cells.  It should reposition them in case they were left out of position by an operation that was stopped by the user.  These are the lines that were there in the previous version:
```
            self.canvas.coords(drawItem.display_shape, *self.cellCoords(i))
            self.canvas.coords(drawItem.display_val, *self.cellCenter(i))
```

Reply:
Fixed. I also noticed, now that you brought it up, that if remove() is paused and stopped halfway through, the position of the nItems pointer gets messed up. Should I add that to fixCells()?

Reply:
That would be great.  The latest update to the Heap.py on `master` has code in it to correct the vertical position of the `nItems` index.  In the PriorityQueue, of course, it would be horizontal.

Reply:
I've tried to implement the system from Heap, but it doesn't work. In addition, I found that if the animation is paused and then stopped too early, insert() may return an error, possibly because nItems wasn't reverted properly.

Reply:
Your implementation was close.  I went ahead and made `fixPositions()` always reset the `nItems` index (not as part of the loop over items in the queue).  I also copied over the logic from Heap.py on when `cleanUp()` calls `fixPositions()`.  It's only needed when cleaning up all of the call stack environments.